### PR TITLE
fix: handle mixed time-series error

### LINKED ISSUE #16834: Mixed Time-Series chart rendering issue
Unexpected error shows when rendering chart occasionally. 

### Expected results

Chart should show the rendering result

### Actual results

Occasionally, chart fails on rendering and shows "An error occurred while rendering the visualization: TypeError: Cannot read properties of undefined (reading 'data')"

#### Screenshots

Screen Shot 2021-09-24 at 8 03 26 PM

#### How to reproduce the bug

1. Update to Superset 1.3 (specific commit_sha: 6d19de8f66e6409cc82d6d1e5db2496c1ca2290d)
2. Create a mixed timeseries chart
3. Render multiple times and check error 

### Environment

(please complete the following information):

- browser type and version: Chrome 93
- superset version: 1.3
- python version: `python --version`
- node.js version: `node -v`
- any feature flags active: GLOBAL_ASYNC_QUERIES

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### PR DESCRIPTION
**Pull request by yougyoung94**
### SUMMARY

Fixes #16834 
https://github.com/apache/superset/issues/16834

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

**Before:**

https://user-images.githubusercontent.com/37649640/135747566-7a281411-ce70-444d-8e1e-4385b3df08ee.mov

**After:**

https://user-images.githubusercontent.com/37649640/135747576-320d22c7-bdba-41ce-a30c-0df0b28d73f7.mov

### TESTING INSTRUCTIONS

1. Apply changes to `superset-frontend/src/chart/chartAction.js`
2. Build image and update `docker-compose-non-dev.yml`
3. Update `docker/pythonpath_dev/superset_config.py` to use GLOBAL_ASYNC_QUERIES and redis cache
4. Run `docker-compose -f docker-compose-non-dev.yml up`
5. Create a mixed time-series chart and check if it gets data from cache correctly

### ADDITIONAL INFORMATION

- [x] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/chart/chartAction.js [hunk 1]
```diff
@@ -394,15 +394,17 @@ export function exploreJSON(
...
+              return Promise.resolve(result);
             case 202:
-              // Query is running asynchronously and we must await the results
-              return waitForAsyncData(result);
+              if (shouldUseLegacyApi(formData)) {
+                // Query is running asynchronously and we must await the results
+                return waitForAsyncData(result[0]);
+              } else {
+                return waitForAsyncData(result);
+              }
```
Comment (geido) [inline superset-frontend/src/chart/chartAction.js:407]:
```suggestion
              if (shouldUseLegacyApi(formData)) {
                // Query is running asynchronously and we must await the results
                return waitForAsyncData(result[0]);
              }
              return waitForAsyncData(result);
```

**Review by geido — APPROVED** [review]
LGTM! Thank you very much for your contribution @yougyoung94. It would be awesome to add a video and some more information for testing. Aside from that, this is great.

File: superset-frontend/src/chart/chartAction.js (hunk 1 above)
Comment (yougyoung94) [reply superset-frontend/src/chart/chartAction.js:407]: Sure! I will upload the video as soon as I can :):)

Comment (yougyoung94) [reply superset-frontend/src/chart/chartAction.js:407]: @geido Thanks for your review:) I uploaded the videos and information for testing:)

**Comment by geido** [comment]
@yougyoung94 it appears that you have some linting issues. I added a suggestion up there for you to solve that problem. Let me know if you need any additional help to make this pass. Thank you!

**Comment by yougyoung94** [comment]
@geido Thanks for your review! I missed that point 😅

**Review by villebro — APPROVED** [review]
LGTM, thank so much for fixing this!

**Comment by villebro** [comment]
@yougyoung94 you need to open this PR against master branch. We'll try to get it into 1.3, but it first needs to hit master and then be picked into 1.3.

**Review by villebro — CHANGES_REQUESTED** [review]
Open PR against master branch

**Comment by yougyoung94** [comment]
@villebro Okay I will:):)

**Comment by yougyoung94** [comment]
@villebro  Done:) LGTM please 🙏

**Review by villebro — APPROVED** [review]

**Pull request merged**